lang/cs/Org.Apache.REEF.IMRU.Examples/PipelinedBroadcastReduce/PipelinedBroadcastAndReduceFaultTolerant.cs

```diff
@@ -0,0 +1,158 @@
+                ////.BindSetEntry<TaskIdsToFail, string>(GenericType<TaskIdsToFail>.Class, "IMRUMap-RandomInputPartition-3-")
+                .BindIntNamedParam<FailureType>("0")
+                .BindNamedParameter(typeof(MaxRetryNumberInRecovery), maxRetryInRecovery.ToString())
+                .Build();
+
+            return Configurations.Merge(c1, c2);
+        }
+
+        [NamedParameter]
+        internal class TaskIdsToFail : Name<ISet<string>>
+        {
+        }
+
+        [NamedParameter(Documentation = "Type of failure to simulate: 0/1 - evaluator/task during task execution, 2/3 - during task initialization")]
+        internal class FailureType : Name<int>
+        {
+        }
+
+        /// <summary>
+        /// The function is to simulate Evaluator/Task failure
+        /// </summary>
+        internal sealed class TestSenderMapFunction : IMapFunction<int[], int[]>
+        {
+            private int _iterations;
+            private readonly string _taskId;
+            private readonly ISet<string> _taskIdsToFail;
+            private int _failureType;
+            private readonly int _maxRetryInRecovery;
+
+            [Inject]
+            private TestSenderMapFunction(
+                [Parameter(typeof(TaskConfigurationOptions.Identifier))] string taskId,
+                [Parameter(typeof(TaskIdsToFail))] ISet<string> taskIdsToFail,
+                [Parameter(typeof(FailureType))] int failureType,
+                [Parameter(typeof(MaxRetryNumberInRecovery))] int maxRetryNumberInRecovery)
+            {
+                _taskId = taskId;
+                _taskIdsToFail = taskIdsToFail;
+                _failureType = failureType;
+                _maxRetryInRecovery = maxRetryNumberInRecovery;
+                Logger.Log(Level.Info, "TestSenderMapFunction: TaskId: {0}, _maxRetryInRecovery {1},  Failure type: {2}.", _taskId, _maxRetryInRecovery, _failureType);
+                foreach (var n in _taskIdsToFail)
+                {
+                    Logger.Log(Level.Info, "TestSenderMapFunction: taskIdsToFail: {0}", n);
+                }
+
+                if (_failureType == 2 || _failureType == 3)
+                {
+                    SimulateFailure(0);
+                }
+            }
+
+            /// <summary>
+            /// Map function
+            /// </summary>
+            /// <param name="mapInput">integer array</param>
+            /// <returns>The same integer array</returns>
+            int[] IMapFunction<int[], int[]>.Map(int[] mapInput)
+            {
+                _iterations++;
+                Logger.Log(Level.Info, "Received value {0} in iteration {1}.", mapInput[0], _iterations);
+
+                if (_failureType <= 1)
+                {
+                    SimulateFailure(10);
+                }
+
+                if (mapInput[0] != _iterations)
+                {
+                    Exceptions.Throw(new Exception("Expected value in mappers different from actual value"), Logger);
```

Comment:
Please add the actual and expected values to the error message to ease debugging.